src/GogeToken.sol

```diff
@@ -596,346 +584,350 @@ contract DogeGaySon is ERC20, Ownable {
         
-        IUniswapV2Router02 _uniswapV2Router = IUniswapV2Router02(0x10ED43C718714eb63d5aA57B78B54704E256024E); // bsc pancakeswap router
+        uniswapV2Router = IUniswapV2Router02(0x10ED43C718714eb63d5aA57B78B54704E256024E);
 
         // Create a uniswap pair for this new token
-        address _uniswapV2Pair = IUniswapV2Factory(_uniswapV2Router.factory()).createPair(address(this), _uniswapV2Router.WETH());
+        address _uniswapV2Pair = IUniswapV2Factory(uniswapV2Router.factory()).createPair(address(this), uniswapV2Router.WETH());
 
-        uniswapV2Router = _uniswapV2Router;
         uniswapV2Pair = _uniswapV2Pair;
 
         _setAutomatedMarketMakerPair(_uniswapV2Pair, true);
         
+        // exclude from paying dividends to the team wallets and dead addresses
         cakeDividendTracker.excludeFromDividends(address(cakeDividendTracker));
-        
         cakeDividendTracker.excludeFromDividends(address(this));
-        cakeDividendTracker.excludeFromDividends(address(_uniswapV2Router));
-        cakeDividendTracker.excludeFromDividends(deadAddress);
+        cakeDividendTracker.excludeFromDividends(address(uniswapV2Router));
+        cakeDividendTracker.excludeFromDividends(DEAD_ADDRESS);
+        cakeDividendTracker.excludeFromDividends(address(0));
         cakeDividendTracker.excludeFromDividends(owner());
+        cakeDividendTracker.excludeFromDividends(devWallet);
+        cakeDividendTracker.excludeFromDividends(marketingWallet);
+        cakeDividendTracker.excludeFromDividends(teamWallet);
 
         // exclude from paying fees or having max transaction amount
         isExcludedFromFees[marketingWallet] = true;
         isExcludedFromFees[teamWallet] = true;
+        isExcludedFromFees[devWallet] = true;
         isExcludedFromFees[address(this)] = true;
         isExcludedFromFees[owner()] = true;
+        isExcludedFromFees[DEAD_ADDRESS] = true;
+        isExcludedFromFees[address(0)] = true;
         
-        /*
-            _mint is an internal function in ERC20.sol that is only called here
-            and for migration, and it CANNOT be called for any other reason. The 
-            external capability does NOT exist.
-        */
         _mint(owner(), _totalSupply * (10**18));
-        //_mint(owner(), 100000000000 * (10**18));
     }
 
     receive() external payable {
 
     }
 
-    function setDAO(address _dao) external {
-        require(_msgSender() == DAO || _msgSender() == owner(), "Not authorized");
-        require(DAO != _dao, "This address is already the DAO");
-        DAO = _dao;
+    function setGogeDao(address _gogeDao) external {
+        require(_msgSender() == gogeDao || _msgSender() == owner(), "GogeToken.sol::setgogeDao() msg.sender is not owner or gogeDao");
+        require(gogeDao != _gogeDao, "GogeToken.sol::setgogeDao() address is already set");
+
+        gogeDao = _gogeDao;
+
+        isExcludedFromFees[gogeDao] = true;
```

Comment:
Should we include the previous dao address for fees on the off chance that address is involved again or used? Not sure how important it is to reset it or if that would be necessary.

Reply:
Good catch, but honestly I think it'd be fine. If we change the DAO address there's 0 interactions that the old DAO should have with this contract. And if tokens are ever transferred to the old contract they'd be in our possession anyway so no reason to tax the transfer